pkg/storage/etcd/etcd_watcher.go

```diff
@@ -398,10 +404,14 @@ func (w *etcdWatcher) sendModify(res *etcd.Response) {
 	var oldObj runtime.Object
 	if res.PrevNode != nil && res.PrevNode.Value != "" {
 		// Ignore problems reading the old object.
-		if oldObj, err = w.decodeObject(res.PrevNode); err == nil {
+		if oldObj, err = w.uncachedDecodeObject(res.PrevNode); err == nil {
 			oldObjPasses = w.filter(oldObj)
 		}
+		if err := w.versioner.UpdateObject(oldObj, res.Node.ModifiedIndex); err != nil {
```

Comment:
I agree with that too. Because both are correct.

etcd2 returns current index and prev index. etcd3 returns only the current index. There is some discussion around to make it return old value & index too.

I prefer to get this fixed and see more use cases in k8s.